go/enclave/limiters/batchlimiter.go

```diff
@@ -0,0 +1,75 @@
+package limiters
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// BatchSizeLimiter - Acts as a limiter for batches based
+// the data from the transaction that we have to publish to the l1.
+// Acts as a calldata reservation system that accounts for both
+// transactions and cross chain messages.
+type batchSizeLimiter struct {
+	Size uint64
+}
+
+type BatchSizeLimiter interface {
+	AcceptTransaction(tx *types.Transaction) error
+	// ProcessReceipt(receipt *types.Receipt) error //todo @stefan add this again
+}
+
+var ErrInsufficientSpace = errors.New("insufficient space in BatchSizeLimiter")
```

Comment:
should also go to the interfaces file, as I assume this is part of the interface